packages/react/src/transactions/infrastructure/PublicationMetadataUploader.ts

```diff
@@ -0,0 +1,26 @@
+import {
+  CreateCommentRequest,
+  CreatePostRequest,
+} from '@lens-protocol/domain/use-cases/publications';
+import { assertError, Url } from '@lens-protocol/shared-kernel';
+
+import { FailedUploadError, IMetadataUploader } from '../adapters/IMetadataUploader';
+import { MetadataUploadHandler } from '../adapters/MetadataUploadHandler';
+import { createPublicationMetadata } from './createPublicationMetadata';
+
+export class PublicationMetadataUploader<T extends CreatePostRequest | CreateCommentRequest>
+  implements IMetadataUploader<T>
+{
+  constructor(private readonly handler: MetadataUploadHandler) {}
+
+  async upload(request: T): Promise<Url> {
+    try {
+      const metadata = createPublicationMetadata(request);
```

Comment:
nit: just API disproportion which can be good,  `PublicationMetadataUploader` takes the `request` whereas the `ProfileMetadataUploader` takes the metadata directly.

Reply:
I know, that's why `T` in `IMetadataUploader<T>` is without restrictions.